src/Castle.Core/DynamicProxy/Contributors/ClassProxySerializableContributor.cs

```diff
@@ -30,28 +31,22 @@ namespace Castle.DynamicProxy.Contributors
 	internal class ClassProxySerializableContributor : SerializableContributor
 	{
 		private readonly bool delegateToBaseGetObjectData;
-		private readonly bool implementISerializable;
 		private ConstructorInfo serializationConstructor;
 		private readonly IList<FieldReference> serializedFields = new List<FieldReference>();
 
 		public ClassProxySerializableContributor(Type targetType, IList<MethodInfo> methodsToSkip, Type[] interfaces,
 		                                     string typeId)
 			: base(targetType, interfaces, typeId)
 		{
-			if (targetType.IsSerializable)
-			{
-				implementISerializable = true;
-				delegateToBaseGetObjectData = VerifyIfBaseImplementsGetObjectData(targetType, methodsToSkip);
-			}
+			Debug.Assert(targetType.IsSerializable, "This contributor is intended for serializable types only.");
```

Comment:
This is fine because of this prior check:

https://github.com/castleproject/Core/blob/9143de2e453347f533df8ebc9129d9cb6e05e5a5/src/Castle.Core/DynamicProxy/Generators/BaseClassProxyGenerator.cs#L184-L189